Sources/SwiftSyntax/gyb_generated/syntax_nodes/SyntaxPatternNodes.swift

```diff
@@ -154,13 +148,41 @@ public struct EnumCasePatternSyntax: PatternSyntaxProtocol, SyntaxHashable {
       unexpectedAfterAssociatedTuple?.raw,
     ]
     let data: SyntaxData = withExtendedLifetime(SyntaxArena()) { arena in
-      let raw = RawSyntax.makeLayout(kind: SyntaxKind.enumCasePattern,
-        from: layout, arena: arena)
+      let raw = RawSyntax.makeLayout(
+        kind: SyntaxKind.enumCasePattern, from: layout, arena: arena,
+        leadingTrivia: leadingTrivia, trailingTrivia: trailingTrivia)
       return SyntaxData.forRoot(raw)
     }
     self.init(data)
   }
 
+  public init(
+    leadingTrivia: Trivia? = nil,
+    _ unexpectedBeforeType: UnexpectedNodesSyntax? = nil,
+    type: MissingTypeSyntax? = nil,
```

Comment:
This is an example of the hack - `Missing*` is used so that clients don't start depending on being able to pass a literal directly (which they could do if this was `TypeSyntax`). I've also removed the literal conformance from `Missing`, since it doesn't make sense to be able to construct missing nodes from a literal (it would currently always fatal error).

Once we support 5.7 as the minimum, we want to remove this initializer and update the previous to use `= Optional<TypeSyntax>.none` instead.

Reply:
I don't quite understand why we need this initializer at all. Shouldn't clients be able to use the initializer that's generic on `<T: TypeSyntaxProtocol>`?

Reply:
Calling the other constructor with the defaulted `type` will result in an error about being unable to infer `T`. We could fix this by specifying `type: T? = Optional<TypeSyntax>.none`, but that fails in Swift 5.6.

Reply:
Oh, this is so stupid.

Could you add your message above as a doc-comment somewhere?

Reply:
I couldn't think of anywhere nice to add it as a doc comment. There is already a comment in `generate_default_initialization`. Is that fine or do you want something else?

Reply:
@ahoppen I've added a doc comment to each of these initializers.